### PR TITLE
JSON layout and ${json-encode}, added EscapeForwardSlash (default false). 

### PR DESCRIPTION
**Pull request by 304NotModified**
fixes https://github.com/NLog/NLog/issues/3580

### CODE REVIEW

**Comment by snakefoot** [comment]
I guess in the premature optimization world, then it could be nice to skip the StringBuilder allocation when handling strings with http-address.

Maybe also fix RequiresJsonEscape:

https://github.com/NLog/NLog/blob/d7b33fddd5087afbfe71ccc1a9eb60b69d0767df/src/NLog/Targets/DefaultJsonSerializer.cs#L673-L682

File: src/NLog/Targets/DefaultJsonSerializer.cs [hunk 1]
```diff
@@ -692,14 +691,19 @@ internal static void AppendStringEscape(StringBuilder destination, string text,
...
+
+        internal static bool RequiresJsonEscape(char ch, bool escapeUnicode, bool escapeForwardSlash)
         {
             if (!EscapeChar(ch, escapeUnicode))
             {
                 switch (ch)
                 {
                     case '"':
-                    case '\\':
+                    case '\\': return escapeForwardSlash;
```
Comment (snakefoot) [inline src/NLog/Targets/DefaultJsonSerializer.cs]: Looks like you chose the backward-slash instead of the forward-slash?

Comment (snakefoot) [reply src/NLog/Targets/DefaultJsonSerializer.cs]: Maybe something like this:
```c#
               switch (ch)
                {
                    case '/': return escapeForwardSlash;
                    case '"':
                    case '\\':
                        return true;
```

Comment (304NotModified) [reply src/NLog/Targets/DefaultJsonSerializer.cs]: oops, thanks!

**Comment by 304NotModified** [comment]
note: default is `true` (backwards comp)

Fixed docs in https://github.com/NLog/NLog/wiki/JsonLayout
and https://github.com/NLog/NLog/wiki/JSON-Encode-Layout-Renderer

**Pull request merged**